crates/red_knot_python_semantic/src/types.rs

```diff
@@ -216,14 +216,45 @@ impl<'db> Type<'db> {
         matches!(self, Type::Unbound)
     }
 
-    pub const fn is_unknown(&self) -> bool {
-        matches!(self, Type::Unknown)
-    }
-
     pub const fn is_never(&self) -> bool {
         matches!(self, Type::Never)
     }
 
+    pub const fn expect_class(self) -> ClassType<'db> {
+        let Type::Class(class_type) = self else {
+            panic!("Expected a class!")
+        };
+        class_type
+    }
+
+    pub const fn expect_module(self) -> File {
+        let Type::Module(file) = self else {
+            panic!("Expected a module!")
+        };
+        file
+    }
+
+    pub const fn expect_union(self) -> UnionType<'db> {
+        let Type::Union(union_type) = self else {
+            panic!("Expected a union!")
+        };
+        union_type
+    }
+
+    pub const fn expect_intersection(self) -> IntersectionType<'db> {
+        let Type::Intersection(intersection_type) = self else {
+            panic!("Expected an intersection!")
+        };
+        intersection_type
+    }
+
+    pub const fn expect_function(self) -> FunctionType<'db> {
+        let Type::Function(function) = self else {
+            panic!("Expected a function!")
+        };
+        function
+    }
```

Comment:
Nit: My preferred way here is:

```
let function =  ty.into_function().unwrap();
```

I do find `into_*` helpers generally useful and I'm fine if tests simply fail with an `unwrap`.